Source/WebCore/rendering/svg/RenderSVGText.cpp

```diff
@@ -284,14 +284,12 @@ void RenderSVGText::subtreeTextDidChange(RenderSVGInlineText* text)
     }
 
     // The positioning elements cache depends on the size of each text renderer in the
-    // subtree. If this changes, clear the cache. It's going to be rebuilt below.
+    // subtree. If this changes, clear the cache. If this changes, clear the cache and mark it for rebuilding in the next layout.
     m_layoutAttributesBuilder.clearTextPositioningElements();
 
     checkLayoutAttributesConsistency(this, m_layoutAttributes);
-    for (RenderObject* descendant = text; descendant; descendant = descendant->nextInPreOrder(text)) {
-        if (is<RenderSVGInlineText>(*descendant))
-            m_layoutAttributesBuilder.buildLayoutAttributesForTextRenderer(downcast<RenderSVGInlineText>(*descendant));
-    }
```

Comment:
What is so special about `subtreeTextDidChange()`? The same function `buildLayoutAttributesForTextRenderer()` is called from `subtreeChildWasAdded()` and  `subtreeChildWasRemoved()` in the same file. So why do not we apply the same fix in these two functions as well?

Also are we sure this will not affect JS querying the SVG text in the same update-the-rendering this function is called?